City-of-Helsinki/drupal-hdbt
# UHF-2870: Hid the result counter from service and unit listings on se…

# Hide result counters from service and unit listings when there is no pager [UHF-2870](https://helsinkisolutionoffice.atlassian.net/browse/UHF-2870)
When services or units are listed on service or unit page or in service list paragraph the result counter should be visible only if there is so many results that the pager is also shown. On the original task there is only service list paragraph mentioned but I think it is important to have consistency throughout the site on all blocks so I extended the functionality similar service and unit lists.

## What was done
* Added a variable for templates that has the defined pagers items per page value. With these values we add a conditional check around the counter on template to see if the total rows of the views results exceeds the pager item per page value. Only if it exceeds the counter is printed.
* Adjusted margin top on the service units and unit services listings to be implemented in similar way and consistently.

## How to install
* On any instance that has TPR-integrations enabled update the hdbt-theme
    * `composer require drupal/hdbt:dev-UHF-2870_hide_counters_if_there_is_no_load_more`
* Run `make drush-cr`

## How to test
* Go to any unit page that has services (for example Oodi) and publish two of the services and also publish the unit.
* Now on the service pages that you just published should be one (if not more) units visible and if there is less than 9 units published then there shouldn't appear a result counter on the listing of units that offer the service.
* Now on the unit page you just published should be two (if not more) services visible and if there is less than 5 services published then there shouldn't appear a result counter on the listing of services that are offered on the unit.
* Next you should try to find a unit with plenty of services (for example health centers) and publish services under the unit.
* Now you guessed it - if there is more than 5 services on the units service listing there comes a pager and the counter as well.
* You can try the same thing with a service with plenty of units (for example library services) and make sure it works the same way.
* Now for the last test go to add a service listing paragraph and add only 4 services on it. There shouldn't be counter or pager - Add more services until the pager and counter appear.

## Other PRs
* Nope.
